src/tests/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -777,6 +778,23 @@ public void testNullHeaderSerialization(final File inputFile) throws Exception {
         Assert.assertEquals(deserializedSAMRecord, initialSAMRecord, "Deserialized SAMRecord not equal to original SAMRecord");
     }
 
+
+    // test to recreate htsjdk issue 316 https://github.com/samtools/htsjdk/issues/316
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testValidateNonsenseCigar(){
+        // Create nonsense record
+        SAMRecord rec = createTestRecordHelper();
+        rec.setCigarString("nonsense");
+
+        //The default validationStringency of a sam record is SILENT.
+        rec.setValidationStringency(ValidationStringency.STRICT);
+        // Validate record
+        List<SAMValidationError> err = rec.validateCigar(-1);
```

Comment:
This test is kind of deceptive.  The test is throwing on this line in `rec.validateCigar(-1)`  So it never actually reaches those asserts below it.  

It looks like this test is proving that `validateCigar()` has a bug.  The contract says that it will `@return List of errors, or null if no errors.`, but instead of doing so it throws an `IllegalArgumentException` if there is a completely invalid cigar.